react_juce/core/EcmascriptEngine_Hermes.cpp

```diff
@@ -0,0 +1,522 @@
+        jsi::Value varToJSIValue(const juce::var &v, jsi::Runtime &runtime);
+
+        juce::var jsiArrayToVarArray(const jsi::Array &v, jsi::Runtime &runtime)
+        {
+            juce::Array<juce::var> varArray;
+
+            const size_t numItems = v.size(runtime);
+            for (size_t i = 0; i < numItems; ++i)
+            {
+                varArray.add(jsiValueToVar(v.getValueAtIndex(runtime, i), runtime));
+            }
+
+            return varArray;
+        }
+
+        juce::var::NativeFunction jsiMethodToVarMethod(jsi::Function v, jsi::Runtime &runtime)
+        {
+            auto fPtr = std::make_shared<CopyableJSIMethodWrapper>(std::move(v));
+
+            return [fPtr = fPtr, &rt = runtime] (const juce::var::NativeFunctionArgs &args)
+            {
+                std::vector<jsi::Value> jsiArgs;
+                for (int i = 0; i < args.numArguments; ++i)
+                {
+                    jsiArgs.emplace_back(varToJSIValue(args.arguments[i], rt));
+                }
+
+                const jsi::Value *jsiArgsPtr = jsiArgs.data();
+                return jsiValueToVar(fPtr->fn.call(rt, jsiArgsPtr, jsiArgs.size()), rt);
+            };
+        }
+
+        juce::var jsiObjectToVarObject(jsi::HostObject &v, jsi::Runtime &runtime)
+        {
+            juce::DynamicObject::Ptr varObj = new juce::DynamicObject();
+
+            auto props = v.getPropertyNames(runtime);
+            for (auto &propID : props)
+            {
+                jsi::Value property = v.get(runtime, propID);
+                varObj->setProperty(juce::String(propID.utf8(runtime)), jsiValueToVar(property, runtime));
+            }
+
+            return varObj.get();
+        }
+
+        juce::var jsiValueToVar(const jsi::Value &v, jsi::Runtime &runtime)
+        {
+            if (v.isBool())
+                return v.getBool();
+
+            if (v.isNumber())
+                return v.getNumber();
+
+            if (v.isString())
+                return juce::String(v.getString(runtime).utf8(runtime));
+
+            if (v.isUndefined())
+                return juce::var::undefined();
+
+            if (v.isNull())
+                return juce::var();
+
+            if (v.isObject())
+            {
+                auto obj = v.getObject(runtime);
+
+                if (obj.isArray(runtime))
+                    return jsiArrayToVarArray(obj.getArray(runtime), runtime);
+
+                if (obj.isFunction(runtime))
+                    return jsiMethodToVarMethod(obj.asFunction(runtime), runtime);
+
+                if (obj.isHostObject(runtime))
+                {
+                    auto hostObjectPtr = obj.getHostObject(runtime);
+                    return jsiObjectToVarObject(*hostObjectPtr, runtime);
+                }
+
+                if (obj.isArrayBuffer(runtime))
+                {
+                    jsi::ArrayBuffer arrayBuffer = obj.getArrayBuffer(runtime);
```

Comment:
IIInnnnteresting. Are we using these array buffers anywhere? Feel like there's probably some cool stuff we could do with array buffers

Reply:
We're not at the moment but yeah I was thinking the same. Could be a nice way to pack data for something expensive you want to send over the bridge etc. Thought I'd implement as it's there. Just need to speak to hermes so we can do the same inverse operation.